### PR TITLE
Release 3.2.2

### LINKED ISSUE #4006: global: make a v3.2.2 release
After having fixed issues during the upcoming sprint, make a new v3.2.2 release of Invenio v3.2 that bumps the min versions of maintained modules.

- Work has to be done on maint-3.2 branch
- Release notes
- Cookiecutter should bump to v3.2.2

### PR DESCRIPTION
**Pull request by ppanero**
Closes #4006

### CODE REVIEW

File: docs/releases/v3.2.2.rst [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+Invenio v3.2.2 fixes issues with incompatibilities with several libraries:
+- WTForms recently release has moved `email_validator` to an optional
+dependency. It has been added in the corresponding packages.
+- SQLAlchemy-Utils v0.36 breaks compatibility of Invenio models with MySQL. It
+enforces setting the length on type `VARCHAR`.
+- Sphinx v3.0+ breaks the build of documentation, due to a Flask context issue.
+- Flask is now provided as a centrally managed dependency from `Invenio-Base`.
+
+In addition Invenio v3.2.2 removes support for Python versions lower than 3.6.0.
```
Comment (lnielsen) [inline docs/releases/v3.2.2.rst:21]:
```suggestion
Invenio v3.2.2 support Python version 3.5 and 3.6. Python 2 support in Invenio ended on January 1st, 2020 with the official end of life for Python 2 on the same date.
```

File: docs/releases/v3.2.2.rst [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+*Released 2020-05-14*
+
+Invenio v3.2.2 fixes issues with incompatibilities with several libraries:
+- WTForms recently release has moved `email_validator` to an optional
+dependency. It has been added in the corresponding packages.
+- SQLAlchemy-Utils v0.36 breaks compatibility of Invenio models with MySQL. It
+enforces setting the length on type `VARCHAR`.
+- Sphinx v3.0+ breaks the build of documentation, due to a Flask context issue.
+- Flask is now provided as a centrally managed dependency from `Invenio-Base`.
```
Comment (lnielsen) [inline docs/releases/v3.2.2.rst:19]: Is this not only in Invenio v3.3?

File: .travis.yml [hunk 3]
```diff
@@ -44,7 +44,6 @@ env:
     - REQUIREMENTS=devel EXTRAS=all,postgresql,elasticsearch7 SQLALCHEMY_DATABASE_URI="postgresql+psycopg2://postgres@localhost:5432/invenio" ES_URL=$ES7_DOWNLOAD_URL DEPLOY=true
 
 python:
-  - "2.7"
```
Comment (lnielsen) [inline .travis.yml:47]: Can you add 3.5 (just because it's declared like that originally in v3.2 release).

File: docs/releases/v3.2.2.rst (hunk 2 above)
Comment (ppanero) [reply docs/releases/v3.2.2.rst:19]: Sadly no, many backports needed this changes to fix the build (Sphinx + Flask). SQLAlchemy-Utils is somehow the same, they release in between and "broke" our compatibility. So the pin needed to be backported (invenio-db v1.0.5, which is compatible with v3.2 and v.3.3)

**Comment by ppanero** [comment]
Have done the quickstart + record and file creation/search and it is working. In addition, there is ongoing work on fixing the tutorials.

- https://invenio.readthedocs.io/en/latest/quickstart/quickstart.html ✅ 
- https://invenio.readthedocs.io/en/latest/quickstart/crud-operations.html#crud-operations ✅

**Comment by ppanero** [comment]
Build is breaking for two reasons:

1-  Cannot find `invenio-app 1.2.6` But it does exists in [PyPi](https://pypi.org/project/invenio-app/1.2.6/)

```
ERROR: Could not find a version that satisfies the requirement invenio-app==1.2.6<1.3.0 (from -r .travis-lowest-requirements.txt (line 7)) (from versions: 1.0.0a1, 1.0.0a2, 1.0.0a3, 1.0.0b1, 1.0.0b2, 1.0.0, 1.0.1, 1.0.2, 1.0.3, 1.0.4, 1.0.5, 1.0.6, 1.1.0, 1.1.1, 1.2.0, 1.2.1, 1.2.2, 1.2.3, 1.2.5, 1.2.6)
1026ERROR: No matching distribution found for invenio-app==1.2.6<1.3.0 (from -r .travis-lowest-requirements.txt (line 7))
```

2- Cannot find CSV serializer in `invenio-records-rest` docs. I can see it has disappeared from the [docs](https://invenio-records-rest.readthedocs.io/en/latest/api.html#csv). Nonetheless, the [latest build](https://travis-ci.org/github/inveniosoftware/invenio-records-rest/builds/684646338) passed and seemed to be fine.

```
/home/travis/build/inveniosoftware/invenio/docs/releases/v3.2.0.rst:159:py:class reference target not found: invenio_records_rest.serializers.csv.CSVSerializer
```

@lnielsen Any ideas?

**Pull request merged**